### PR TITLE
Use empty string as contract address for ETH

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/18487
Resolves https://github.com/brave/brave-browser/issues/18454

This PR update the default value of user assets pref to use empty string as ETH's contract address.
And also migrate the pref to reset ETH's contract address to an empty string from all stored assets.

@Douglashdaniel also fixed front-end usage where we sometimes use `eth` as the contract address in this PR.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by yrliou** [comment]
C++ side work is ready for review, but we'll have to push front-end fix in this PR too otherwise it would break desktop wallet frontend.

File: components/brave_wallet/browser/pref_names.cc [hunk 1]
```diff
@@ -16,6 +16,8 @@ const char kBraveWalletCurrentChainId[] =
     "brave.wallet.wallet_current_chain_id";
 const char kBraveWalletKeyrings[] = "brave.wallet.keyrings";
 const char kBraveWalletUserAssets[] = "brave.wallet.user_assets";
+const char kBraveWalletUserAssetEthContractAddressMigrated[] =
+    "brave.wallet.user.asset.eth.contract.address.migrated";
```
Comment (darkdh) [inline components/brave_wallet/browser/pref_names.cc:20]: brave.wallet.user_asset.eth_contract_address_migrated

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 2]
```diff
@@ -111,6 +110,10 @@ void BraveWalletService::Bind(
 absl::optional<std::string> BraveWalletService::GetChecksumAddress(
     const std::string& contract_address,
     const std::string& chain_id) {
+  if (contract_address.empty()) {
```
Comment (darkdh) [inline components/brave_wallet/browser/brave_wallet_service.cc:113]: shouldn't we prevent assets other than eth from adding empty contract_address?

Comment (darkdh) [reply components/brave_wallet/browser/brave_wallet_service.cc:113]:
```
diff --git a/components/brave_wallet/browser/brave_wallet_service.cc b/components/brave_wallet/browser/brave_wallet_service.cc
index 36adf975dd..6e2a64c641 100644
--- a/components/brave_wallet/browser/brave_wallet_service.cc
+++ b/components/brave_wallet/browser/brave_wallet_service.cc
@@ -214,6 +214,10 @@ void BraveWalletService::AddUserAsset(mojom::ERCTokenPtr token,
     std::move(callback).Run(false);
     return;
   }
+  if (optional_checksum_address->empty() && token->symbol != "ETH") {
+    std::move(callback).Run(false);
+    return;
+  }
   const std::string checksum_address = optional_checksum_address.value();

   const std::string network_id = GetNetworkId(prefs_, chain_id);

```

File: components/brave_wallet/browser/pref_names.cc (hunk 1 above)
Comment (yrliou) [reply components/brave_wallet/browser/pref_names.cc]: Fixed in https://github.com/brave/brave-core/pull/10333/commits/75a3d3253f5b0e0d65f6e163616dce34f5c69756

File: components/brave_wallet/browser/brave_wallet_service.cc (hunk 2 above)
Comment (yrliou) [reply components/brave_wallet/browser/brave_wallet_service.cc:113]: Fixed in https://github.com/brave/brave-core/pull/10333/commits/970f83b682abfe55a9ff473a9d5af166c8ec5130

**Review by darkdh — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
`chromium_src` changes ++

**Review by yrliou — COMMENTED** [review]
Last commit authored by @Douglashdaniel for fixing front end usage looks good to me.

**Pull request merged**